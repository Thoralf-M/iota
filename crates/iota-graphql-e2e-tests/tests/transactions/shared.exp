processed 7 tasks

task 1, lines 7-23:
//# publish
created: object(1,0), object(1,1)
mutated: object(0,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 6847600,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, lines 25-26:
//# programmable --inputs immshared(1,0)
//> 0: P0::m::get(Input(0))
mutated: object(0,0)
unchanged_shared: object(1,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 980400,  storage_rebate: 980400, non_refundable_storage_fee: 0

task 3, lines 28-29:
//# programmable --inputs object(1,0)
//> 0: P0::m::inc(Input(0))
mutated: object(0,0), object(1,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 2257200,  storage_rebate: 2257200, non_refundable_storage_fee: 0

task 4, lines 31-33:
//# programmable --inputs object(1,0)
//> 0: P0::m::get(Input(0));
//> P0::m::inc(Input(0))
mutated: object(0,0), object(1,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 2257200,  storage_rebate: 2257200, non_refundable_storage_fee: 0

task 5, line 35:
//# create-checkpoint
Checkpoint created: 1

task 6, lines 37-85:
//# run-graphql
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "kind": {
            "__typename": "ProgrammableTransactionBlock",
            "transactions": {
              "nodes": [
                {
                  "package": "0x0f785d8315ad19c0ad6f7415b518d775e963dc5457df45e92c3ba2fcca766537",
                  "module": "m",
                  "functionName": "get"
                }
              ]
            }
          },
          "effects": {
            "status": "SUCCESS",
            "unchangedSharedObjects": {
              "nodes": [
                {
                  "__typename": "SharedObjectRead",
                  "address": "0x6f07b78ba9141b1a7a6b729880fc9b63b12e557313134c722f2b04169c853a1a",
                  "version": 2,
                  "digest": "bLhuLwagCRxcGmFWMCg9GDDziJhA9u5zqKfpQ81Tn2g",
                  "object": {
                    "asMoveObject": {
                      "contents": {
                        "type": {
                          "repr": "0x0f785d8315ad19c0ad6f7415b518d775e963dc5457df45e92c3ba2fcca766537::m::Foo"
                        },
                        "json": {
                          "id": "0x6f07b78ba9141b1a7a6b729880fc9b63b12e557313134c722f2b04169c853a1a",
                          "x": "0"
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        },
        {
          "kind": {
            "__typename": "ProgrammableTransactionBlock",
            "transactions": {
              "nodes": [
                {
                  "package": "0x0f785d8315ad19c0ad6f7415b518d775e963dc5457df45e92c3ba2fcca766537",
                  "module": "m",
                  "functionName": "inc"
                }
              ]
            }
          },
          "effects": {
            "status": "SUCCESS",
            "unchangedSharedObjects": {
              "nodes": []
            }
          }
        },
        {
          "kind": {
            "__typename": "ProgrammableTransactionBlock",
            "transactions": {
              "nodes": [
                {
                  "package": "0x0f785d8315ad19c0ad6f7415b518d775e963dc5457df45e92c3ba2fcca766537",
                  "module": "m",
                  "functionName": "get"
                },
                {
                  "package": "0x0f785d8315ad19c0ad6f7415b518d775e963dc5457df45e92c3ba2fcca766537",
                  "module": "m",
                  "functionName": "inc"
                }
              ]
            }
          },
          "effects": {
            "status": "SUCCESS",
            "unchangedSharedObjects": {
              "nodes": []
            }
          }
        }
      ]
    }
  }
}
