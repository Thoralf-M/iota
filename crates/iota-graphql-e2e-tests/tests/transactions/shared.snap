---
source: external-crates/move/crates/move-transactional-test-runner/src/framework.rs
---
processed 7 tasks

task 1, lines 7-23:
//# publish
created: object(1,0), object(1,1)
mutated: object(0,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 6847600,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, lines 25-26:
//# programmable --inputs immshared(1,0)
//> 0: P0::m::get(Input(0))
mutated: object(0,0)
unchanged_shared: object(1,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 980400,  storage_rebate: 980400, non_refundable_storage_fee: 0

task 3, lines 28-29:
//# programmable --inputs object(1,0)
//> 0: P0::m::inc(Input(0))
mutated: object(0,0), object(1,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 2257200,  storage_rebate: 2257200, non_refundable_storage_fee: 0

task 4, lines 31-33:
//# programmable --inputs object(1,0)
//> 0: P0::m::get(Input(0));
//> P0::m::inc(Input(0))
mutated: object(0,0), object(1,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 2257200,  storage_rebate: 2257200, non_refundable_storage_fee: 0

task 5, line 35:
//# create-checkpoint
Checkpoint created: 1

task 6, lines 37-85:
//# run-graphql
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "kind": {
            "__typename": "ProgrammableTransactionBlock",
            "transactions": {
              "nodes": [
                {
                  "package": "0x84d71c058bc8a314b1de14879cb4cbc723fa7f8dc28e31c95021e204a7d80c58",
                  "module": "m",
                  "functionName": "get"
                }
              ]
            }
          },
          "effects": {
            "status": "SUCCESS",
            "unchangedSharedObjects": {
              "nodes": [
                {
                  "__typename": "SharedObjectRead",
                  "address": "0x7b85ba777761557bc6824eae0b5bc9d79b71925dbe217a2c2c4b3ae307bf0c01",
                  "version": 2,
                  "digest": "FGhGfSsEbAQjVB8dB6tzYJpUZ4Udtd7y6n2iBqdBemsB",
                  "object": {
                    "asMoveObject": {
                      "contents": {
                        "type": {
                          "repr": "0x84d71c058bc8a314b1de14879cb4cbc723fa7f8dc28e31c95021e204a7d80c58::m::Foo"
                        },
                        "json": {
                          "id": "0x7b85ba777761557bc6824eae0b5bc9d79b71925dbe217a2c2c4b3ae307bf0c01",
                          "x": "0"
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        },
        {
          "kind": {
            "__typename": "ProgrammableTransactionBlock",
            "transactions": {
              "nodes": [
                {
                  "package": "0x84d71c058bc8a314b1de14879cb4cbc723fa7f8dc28e31c95021e204a7d80c58",
                  "module": "m",
                  "functionName": "inc"
                }
              ]
            }
          },
          "effects": {
            "status": "SUCCESS",
            "unchangedSharedObjects": {
              "nodes": []
            }
          }
        },
        {
          "kind": {
            "__typename": "ProgrammableTransactionBlock",
            "transactions": {
              "nodes": [
                {
                  "package": "0x84d71c058bc8a314b1de14879cb4cbc723fa7f8dc28e31c95021e204a7d80c58",
                  "module": "m",
                  "functionName": "get"
                },
                {
                  "package": "0x84d71c058bc8a314b1de14879cb4cbc723fa7f8dc28e31c95021e204a7d80c58",
                  "module": "m",
                  "functionName": "inc"
                }
              ]
            }
          },
          "effects": {
            "status": "SUCCESS",
            "unchangedSharedObjects": {
              "nodes": []
            }
          }
        }
      ]
    }
  }
}
